x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/esql/action/EsqlQueryRequest.java

```diff
@@ -21,6 +21,8 @@ protected EsqlQueryRequest(StreamInput in) throws IOException {
         super(in);
     }
 
+    public abstract String esqlVersion();
```

Comment:
Is this meant to be the unparsed string? If so, I'm not sure there's merit in exposing that instead of the enum constant.

Reply:
Discussed: having just the unparsed string here, rather than an `EsqlVersion` object, avoids having to serialize the latter. Parsing this should be cheap, so we can re-parse it e.g. in `EsqlSession.execute()` and turn it into appropriate behavior flags there.